src/particles/PhysicsParticles.hpp

```diff
@@ -642,44 +695,67 @@ template <typename problem_t> class PhysicsParticleRegister
 		return false;
 	}
 
-	// TODO(cch): replace name with particleType to get rid of the string comparison
+	// Utility method to convert particle type to string name (for writing plotfiles/checkpoints)
+	[[nodiscard]] static auto getParticleTypeName(ParticleType type) -> std::string
+	{
+		switch (type) {
+			case ParticleType::Rad:
+				return "Rad_particles";
+			case ParticleType::CIC:
+				return "CIC_particles";
+			case ParticleType::CICRad:
+				return "CICRad_particles";
+			default:
+				return "Unknown_particles";
+		}
+	}
+
 	// Register a new particle type with specified properties
 	template <typename ContainerType>
-	void registerParticleType(const std::string &name, int mass_idx, int lum_idx, int birth_time_idx, bool hydro_interact, ContainerType *container)
+	void registerParticleType(ParticleType type, int mass_idx, int lum_idx, int birth_time_idx, bool hydro_interact, bool allows_creation,
+				  ContainerType *container)
 	{
 		std::unique_ptr<PhysicsParticleDescriptorBase> descriptor;
-		if (name == "Rad_particles") {
-			descriptor = std::make_unique<PhysicsParticleDescriptor<ContainerType, problem_t, ParticleType::Rad>>(mass_idx, lum_idx, birth_time_idx,
-															      hydro_interact, container);
-		}
+
+		// Create the appropriate descriptor based on the particle type
+		switch (type) {
+			case ParticleType::Rad:
+				descriptor = std::make_unique<PhysicsParticleDescriptor<ContainerType, problem_t, ParticleType::Rad>>(
+				    mass_idx, lum_idx, birth_time_idx, hydro_interact, allows_creation, container);
+				break;
```

Comment:
I would prefer to avoid `switch`/`case` in Quokka in favor of using `if` / `else if`, because it is easy to forget the `break` statement after any one of the cases. If a developer forgets this, the behavior is almost always wrong.